bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/OmnilinkBridgeHandler.java

```diff
@@ -568,21 +568,13 @@ public void dispose() {
     }
 
     private Optional<Thing> getUnitThing(int unitId) {
-        Optional<Thing> theThing = getChildThing(THING_TYPE_UNIT_UPB, unitId);
-        if (!(theThing.isPresent())) {
-            theThing = getChildThing(THING_TYPE_ROOM, unitId);
-        }
-        if (!(theThing.isPresent())) {
-            theThing = getChildThing(THING_TYPE_FLAG, unitId);
-        }
-        if (!(theThing.isPresent())) {
-            theThing = getChildThing(THING_TYPE_OUTPUT, unitId);
-        }
-        if (!(theThing.isPresent())) {
-            theThing = getChildThing(THING_TYPE_DIMMABLE, unitId);
-        }
-        if (!(theThing.isPresent())) {
-            theThing = getChildThing(THING_TYPE_UNIT, unitId);
+        Optional<Thing> theThing = Optional.empty();
+
+        for (ThingTypeUID uid : SUPPORTED_UNIT_TYPES_UIDS) {
+            theThing = getChildThing(uid, unitId);
+            if (theThing.isPresent()) {
+                return theThing;
+            }
```

Comment:
You can go even further and do
```suggestion
        return SUPPORTED_UNIT_TYPES_UIDS.stream().map(uid -> getChildThing(uid, unitId)).flatMap(Optional::stream)
                .findFirst();
```